pkg/promtail/targets/filetarget.go

```diff
@@ -198,12 +159,104 @@ func (t *FileTarget) run() {
 			}
 		case err := <-t.watcher.Errors:
 			level.Error(t.logger).Log("msg", "error from fswatch", "error", err)
+		case <-ticker.C:
+			err := t.sync()
+			if err != nil {
+				level.Error(t.logger).Log("msg", "error running sync function", "error", err)
+			}
 		case <-t.quit:
 			return
 		}
 	}
 }
 
+func (t *FileTarget) sync() error {
+
+	// Find list of directories to add to watcher.
+	var err error
+	path, err := filepath.Abs(t.path)
+	if err != nil {
+		return errors.Wrap(err, "filepath.Abs")
+	}
+
+	// Gets current list of files to tail.
+	matches, err := filepath.Glob(path)
+	if err != nil {
+		return errors.Wrap(err, "filepath.Glob")
+	}
+
+	// Get the current unique set of dirs to watch.
+	dirs := make(map[string]struct{})
```

Comment:
We like to avoid make and prefer the literal style initialisation (`map[string]struct{}{}`), unless you already know the size, in which case you have to use `make`.